libs/auth/src/drivers/magento/login.service.spec.ts

```diff
@@ -0,0 +1,130 @@
+import { DaffAccountRegistration } from '../../models/account-registration';
+import { DaffAuthQueryManager } from '../injection-tokens/auth-query-manager.token';
+import { DaffAuthToken } from '../../models/auth-token';
+import { DaffAuthTransformer } from '../injection-tokens/auth-transformer.token';
+import { DaffCustomerRegistration } from '../../models/customer-registration';
+import { DaffLoginInfo } from '../../models/login-info';
+import { GenerateTokenResponse } from './models/outputs/generate-token-response'
+
+describe('Driver | Magento | Auth | LoginService', () => {
+  let controller: ApolloTestingController;
+  let loginService: DaffMagentoLoginService<
+    DaffLoginInfo,
+    DaffAuthToken,
+    DaffCustomerRegistration,
+    DaffAccountRegistration<DaffCustomerRegistration>,
+    GenerateTokenResponse
+  >;
+
+  const transformerServiceSpy = jasmine.createSpyObj('DaffMagentoAuthTransformerService', ['transform'])
+
+  const registrationFactory: DaffAccountRegistrationFactory = new DaffAccountRegistrationFactory();
+  const authTokenFactory: DaffAuthTokenFactory = new DaffAuthTokenFactory();
+
+  let mockAuth: DaffAuthToken;
+  let mockLoginInfo: DaffLoginInfo;
+  let token: string;
+  let email: string;
+  let password: string;
+  let firstName: string;
+  let lastName: string;
+  let mockRegistration: DaffAccountRegistration<DaffCustomerRegistration>;
+
+  let authGraphQlQueryManagerService: DaffMagentoAuthGraphQlQueryManagerService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        ApolloTestingModule
+      ],
+      providers: [
+        DaffMagentoLoginService,
+        {
+          provide: DaffAuthQueryManager,
+          useExisting: DaffMagentoAuthGraphQlQueryManagerService
+        },
+        {
+          provide: DaffAuthTransformer,
+          useValue: transformerServiceSpy
+        }
+      ]
+    });
+
+    loginService = TestBed.get(DaffMagentoLoginService);
+    controller = TestBed.get(ApolloTestingController);
+    authGraphQlQueryManagerService = TestBed.get(DaffMagentoAuthGraphQlQueryManagerService);
+
+    mockRegistration = registrationFactory.create();
+    mockAuth = authTokenFactory.create();
+
+    token = mockAuth.token;
+    firstName = mockRegistration.customer.firstName;
+    lastName = mockRegistration.customer.lastName;
+    email = mockRegistration.customer.email;
+    password = mockRegistration.password;
+    mockLoginInfo = {email, password};
+  });
+
+  it('should be created', () => {
+    expect(loginService).toBeTruthy();
+  });
+
+  describe('login | getting an access token', () => {
+    let response;
+
+    afterEach(() => {
+      controller.verify();
+    });
+
+    beforeEach(() => {
+      response = {
+        generateCustomerToken: {
+          token
+        }
+      };
+
+      transformerServiceSpy.transform.withArgs(response).and.returnValue(mockAuth);
+    });
+
+    // TODO: determine if mocked tests are appropriate
```

Comment:
You could do this if you want. Maybe do `expect(transformerServiceSpy.transform).toHaveBeenCalledWith(response.generateCustomerToken);` instead if you do though.

Reply:
good advice